lib/datadog/appsec/rate_limiter.rb

```diff
@@ -1,35 +1,14 @@
 # frozen_string_literal: true
 
+require_relative '../core/rate_limiter'
+
 module Datadog
   module AppSec
-    # Simple per-thread rate limiter
-    # Since AppSec marks sampling to keep on a security event, this limits the flood of egress traces involving AppSec
+    # Per-thread rate limiter based on token bucket rate limiter.
+    #
+    # Since AppSec marks sampling to keep on a security event, this limits
+    # the flood of egress traces involving AppSec
```

Comment:
Also, for what it's worth, note that `Thread#[]` in Ruby is not thread-local, or more specifically, it is fiber-local, so you can have N values for a single thread, if multiple fibers are in use. (But if no fibers are in use, you'll have 1 value:1 thread, e.g. thread-local-like behavior).

There's a different API in Ruby for actual thread-locals that are always 1:1 with threads, even in the presence of multiple fibers.

Reply:
My bad, going to update the code to use Thread-local variable set/get

Reply:
Fixed https://github.com/DataDog/dd-trace-rb/pull/3975/commits/e59589647bbbffb56e373df037596f084153cce2